src/battle.ts

```diff
@@ -1187,10 +1196,12 @@ class Battle {
 	}
 	switchSides() {
 		this.setSidesSwitched(!this.sidesSwitched);
+		this.scene.updateSidebars();
 		this.resetToCurrentTurn();
 	}
 	setSidesSwitched(sidesSwitched: boolean) {
 		this.sidesSwitched = sidesSwitched;
+		if (this.mySide === this.p3 || this.mySide === this.p4) return;
```

Comment:
I think switching sides should still be supported if the current side is `p3` or `p4`. What's the meaning of disabling it?

Reply:
If the current side is `p3` or `p4` then we're not a spectator, and active players should never be switching sides except when they first join as a player. Spectator perspectives are always `p1` or `p2`.

Reply:
There's no reason why we shouldn't support active players switching sides. I think it's bad to couple them like this.

Reply:
There's a lot of code that assumes `mySide` will always be `nearSide` and that shouldn't be the assumption; we made them separate variables to support cases where they're not.